serial_communication/serial_bridge/slave/slave_bridge_impl.hpp

```diff
@@ -0,0 +1,107 @@
+// Implementation of slave_bridge.hpp
```

Comment:
Since templated functions must be generated at compile time, their definitions must be available from `#include` statements; otherwise the compiler has no way of knowing how to create them. So all of the code is in header files, but I separated out the implementation and included it from the bottom of slave_bridge.hpp